File: spring-java/src/main/java/com/jmpark/springjava/config/ErrorHandler.java
package com.jmpark.springjava.config;

import com.jmpark.springjava.dto.ErrorException;
import com.jmpark.springjava.dto.ResponseDto;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
public class ErrorHandler {
    @ExceptionHandler(ErrorException.class)
    public ResponseDto<Boolean> exception(ErrorException exception) {
        return ResponseDto.generate(exception.getCode(), "error", false);
    }
}

File: spring-kotlin/src/main/kotlin/com/jmpark/springkotlin/config/ErrorHandler.kt
package com.jmpark.springkotlin.config

import com.jmpark.springkotlin.dto.ErrorException
import com.jmpark.springkotlin.dto.ResponseDto
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class ErrorHandler {
    @ExceptionHandler(ErrorException::class)
    fun exception(exception: ErrorException): ResponseDto {
        return ResponseDto.generate(exception.code, "error", false)
    }
}